Source/GSImageMagickImageRep.m

```diff
@@ -139,6 +143,14 @@ + (NSArray*) imageRepsWithData: (NSData *)data allImages: (BOOL)allImages
   QueryColorDatabase("none", &imageinfo->background_color, exception);
 #endif
 
+  memset(signature, 0, 32);
+  [data getBytes: signature range: NSMakeRange([data length] - 18, 18)];
+  if (strncmp(signature, "TRUEVISION-XFILE.", 17) == 0)
+    {
+      NSWarnLog(@"Targa file detected!, giving a magick hint...");
+      strcpy(imageinfo->magick, "TGA");
```

Comment:
Although it is pretty save here, we normally don't use `strcpy`. Rather something like `strncpy`. Here it makes no difference but static code analysis tools may complain.

Reply:
> Although it is pretty save here, we normally don't use `strcpy`. Rather something like `strncpy`. Here it makes no difference but static code analysis tools may complain.

is it needed even if the right value is a constant? looks overkill, a code analyzer shouldn't complain.